packages/gatsby/src/commands/build.js

```diff
@@ -184,6 +215,35 @@ module.exports = async function build(program: BuildArgs) {
   }
   activity.done()
 
+  let deletedPageKeys = []
+  if (process.env.GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES) {
+    activity = report.activityTimer(`Delete previous page data`)
+    activity.start()
+    deletedPageKeys = pageDataUtil.collectRemovedPageData(
+      store.getState(),
+      cachedPageData
+    )
+
+    const removePages = deletedPageKeys.map(value => {
+      if (value === `/`) {
+        return fs.remove(`${program.directory}/public/index.html`)
+      } else {
+        return fs.remove(`${program.directory}/public${value}`)
+      }
+    })
+    const removePageData = deletedPageKeys.map(value => {
+      if (value === `/`) {
+        return fs.remove(`${program.directory}/public/page-data/index`)
+      } else {
+        return fs.remove(`${program.directory}/public/page-data${value}`)
+      }
```

Comment:
This in practice reimplements https://github.com/gatsbyjs/gatsby/blob/7d843d235fa74be40eda90eb40360a98b91660d8/packages/gatsby/src/utils/page-data.js#L5-L8

Maybe we could add `remove` function in `utils/page-data` (that seems to fit nicely with other function that this file is exporting (`read`, `write`, .... plus hopefully `remove`) and import and use it here?

Reply:
By "this" I mean `if / else` cases and generating paths. If we had `remove` function in `utils/page-data.js`, then we could just reuse `getFilePath` instead of reimplementing it

Reply:
I've reused `getFilePath` but I've had to make a slight change to the function to also return the Html path, not just the page-data path.